### PR TITLE
`Quiz exercises`: Fix adding file attachments on the edit view

### PR DESCRIPTION
**Pull request by sarpsahinalp**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
### Motivation and Context

In any Quiz exercise currently, due to non-unique IDs for the input elements, the inserted file always gets added to the first markdown-editor component if there are multiple ones.

### Description

I changed the IDs of input elements in the MarkdownEditorComponent to ensure uniqueness. This allows selecting the correct editor with a click. Additionally, I added tests for the MarkdownEditorComponent's onFileUpload method.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with quiz exercises

1. Create a new quiz
2. Add two questions
3. On the second question, click "Attach files by dropping or selecting them" (bottom border of the markdown editor)
4. Select an image file and confirm
5. Observe that the link gets added to the second question instead of the first one

#### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 1 Course with quiz exercises

1. Create a new quiz
2. Add two questions
3. On the second question, click "Attach files by dropping or selecting them" (bottom border of the markdown editor)
4. Select an image file and confirm
5. Observe that the link gets added to the second question instead of the first one

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [x] Test 1
- [ ] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| markdown-editor.component.ts | 58.5% | ✅              |

### Screenshots

Before:
https://github.com/ls1intum/Artemis/assets/92453703/689f08a3-d6c4-48ac-a021-48e3bbd5f62b

After: The upload is added in the selected MarkdownEditor
image

## Summary by CodeRabbit

- **New Features**
	- Improved file upload in the Markdown Editor with dynamic IDs for better usability and accessibility.

- **Tests**
	- Implemented unit tests for the Markdown Editor to verify file embedding functionality.

### CODE REVIEW

File: src/test/javascript/spec/component/markdown-editor/markdown-editor.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should create the component', () => {
+        expect(component).toBeTruthy();
+    });
+
```
Comment (Strohgelaender) [inline src/test/javascript/spec/component/markdown-editor/markdown-editor.component.spec.ts:45]: Tests like this basically cannot fail and are therefore meaningless.
```suggestion
```

File: src/test/javascript/spec/component/markdown-editor/markdown-editor.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            },
+        };
+
+        const embedFilesSpy = jest.spyOn(component, 'embedFiles').mockImplementation();
+
+        // Act
+        component.onFileUpload(event as any);
+
+        // Assert
+        expect(embedFilesSpy).toHaveBeenCalledWith([event.target.files[0]]);
```
Comment (Strohgelaender) [inline src/test/javascript/spec/component/markdown-editor/markdown-editor.component.spec.ts:60]:
```suggestion
        const files = [new File([], 'file1.png')];
        const event = {
            target: {
                files: files,
            },
        };

        const embedFilesSpy = jest.spyOn(component, 'embedFiles').mockImplementation();

        // Act
        component.onFileUpload(event as any);

        // Assert
        expect(embedFilesSpy).toHaveBeenCalledWith(files);
```

**Review by Strohgelaender — COMMENTED** [review]
Thanks for having a look at the issue. The changes make sense, I have some small improvemetns to the test code:

Also please make sure to follow [our naming guide for pull requests](https://docs.artemis.cit.tum.de/dev/development-process#naming-conventions-for-github-pull-requests).

**Review by edkaya — APPROVED** [review]
tested on ts2, works as expected

**Review by florian-glombik — APPROVED** [review]
Tested on ts2 by attaching a `.png` file to the second of two multiple choice questions - the bug does not exist anymore 👍 

The code changes are looking good aswell, the id variable name could be changed, but this is probably a matter of taste anyways.

**Comment by florian-glombik** [comment]
Another short notice, we usually do not include the Issue tag in the branch as we link the branches to PRs _(which we link to issues)_ anyways :)

**Review by Strohgelaender — APPROVED** [review]
code lgtm

**Review by egekurt123 — APPROVED** [review]
Manually tested on ts2, works as expected

**Review by az108 — APPROVED** [review]
works

**Review by coolchock — APPROVED** [review]
code lgtm

**Pull request merged**